### ISSUE TITLE
Cannot create dishes

### ISSUE THREAD

**Issue by madmini** [elsantner/canteenMgmtBackend#19]
Trying to create a dish using the `/dish` route with POST does not work

Tested with the following request:

```
curl -i -X POST -H 'Content-Type: application/json' -d '{"name":"spaghet","price":10.0,"type":1}' http://localhost:8080/dish
```

At 75f2ee6572fbad11dedf5119f951cf677f4c0e61 (latest at the time of writing) I get

```
...
{"timestamp":...,"status":403,"error":"Forbidden","path":"/dish"}
```

And the server logs `Invalid CSRF token found for http://localhost:8080/dish`, which is expected, but I do not know how to request such a token from the server.

At cdf1dc0e2d829da1a53dc89dda245c9f177344b8 (latest without CSRF) I get

```
{"timestamp":...,"status":415,"error":"Unsupported Media Type","path":"/dish"}
```

And the server logs `Content type 'application/x-www-form-urlencoded;charset=UTF-8' not supported`.

And at ea981ea7a70b793a3f7bfecddc4c67d98cc45b04 (latest on main, switching https off and adjusting db access) I get `..."status":403,"error":"Forbidden"` again, with no logged error.

As for the frontend, I only receive `XMLHttpRequest error` for all of those.

**Comment by elsantner** [comment]
I noticed that as well. It seems that this is an issue with CORS as well as CSRF. 
I've therefore now enabled CORS for Spring Security (in the most recent build in dev pull request).

Furthermore, now that CSRF is enabled we need to manage the CSRF token. This token is returned after every request in the `Set-Cookie` header. For all state-changing request (POST, PUT, DELETE, ...), this token must be passed in the HTTP header `X-XSRF-TOKEN`. Doing so should result in working POST requests.

However, I've also tried to implement this in the frontend and faced some issues. Firstly, while my implementation seems to work fine while running both frontend and backend locally, CORS issues arrise again when trying to use the Heroku backend with a local frontend. This is further complicated because Cookies are managed by the browser when using Flutter web.

Since the token must be set as the specific request header `X-XSRF-TOKEN`, the frontend needs to access the cookie. Doing so via `html.window.document.cookie` in Flutter works for non-CORS environments. However, Cookies are not set by the browser if they are set in a CORS environment without `SameSite=None` - which is difficult to configure in Spring.
Also, accessing `html.window.document.cookie` breaks the Unit Tests in the frontend. 

I've yet to find a solution for this issue. I've also considered just disabling CSRF protection altogether. However, I suspect we'll run into a similar issue when implementing login / authentication since we most likely also need to set Cookies there...

The (somewhat broken) frontend implementation is available in the frontend branch [feature/csrf-support](https://github.com/AAU-ASE-GroupC-WS2021/canteenMgmtFrontend/tree/feature/csrf-support).
